File: server/services/battle/battle_state.dart
import 'battle.dart';

class BattleState {

  BattleState({
    required this.base,
    required this.finale,
    required this.wall,
    required this.immensity,
    required this.ratio,
  });

  BattleState.defaultValues()
      : base = BattleSides<double>(0, 0),
        finale = BattleSides<double>(0, 0),
        wall = 0,
        immensity = 0.0,
        ratio = 0;
  BattleSides<double> base;
  final BattleSides<double> finale;
  int wall;
  double immensity;
  final int ratio;

  double getRatio() {
    return finale.off / finale.def;
  }
}

File: java_server/src/main/java/xyz/qruto/java_server/models/battle/BattleState.java
package xyz.qruto.java_server.models.battle;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class BattleState {
    private BattleSides<Double> base;
    private BattleSides<Double> finale;
    private int wall;
    private double immensity;
    private int ratio;

    public BattleState() {
        this.base = BattleSides.off(0.0, 0.0);
        this.finale = BattleSides.off(0.0, 0.0);
    }

    public double getRatio(){
        return finale.getOff() / finale.getDef();
    }
}
